sdk/python/kfp_tekton/compiler/_k8s_helper.py

```diff
@@ -46,7 +47,9 @@ def sanitize_k8s_name(name,
     k8s_name = re.sub('[^-_./0-9A-Za-z]+', '-', name)
 
     if not allow_capital_underscore:
```

Comment:
Hi @Tomcli -- since only kfp-tekton is this helper method, should we use this opportunity to split the flag `allow_capital_underscore` into two separate flags `allow_underscore` and `allow_capital`?

Reply:
Also, if we are using `allow_Capital=True` everywhere (or in most places), should we make that the default?

Reply:
The reason I kept the `allow_capital_underscore` flag is to make sure this helper function is still backward compatible with all the existing features we have. For k8s related name we must use lower case and non-underscore, but there's isn't a case yet where we allow underscore but not upper case characters.

Reply:
I saw several places with `sanitize_k8s_name(name, True)`. Maybe refactoring can be a different PR?

Reply:
Okay. Thanks Tommy!